### PR TITLE
How to submit a demo

### PR DESCRIPTION
**Pull request by thisac**
_This is PR 2/3 for the new QML website redesign. The last one being adding related-demos sections to the demos (and populating them with related demos)._

I thought this PR would contain more than it currently does (e.g. a process for a user to actually submit demos/information/links via some sort of form), but it was decided to keep this quite simple. This PR thus simply adds a guide for how to submit a demo via a GitHub issue to the QML repository.

### CODE REVIEW

File: demos_submission.rst [hunk 1]
```diff
@@ -8,8 +8,101 @@ How to submit a demo
    :property="og:description": Take a deeper dive into quantum machine learning by exploring cutting-edge algorithms using PennyLane and near-term quantum hardware.
    :property="og:image": https://pennylane.ai/qml/_static/demos_card.png
 
+To submit your own demo you can follow the steps below, beginning with creating
+a demo, followed by filling in a demo-submission template and posting it as an
+issue to the `PennyLane QML GitHub repository <https://github.com/PennyLaneAI/qml>`_
```
Comment (josh146) [inline demos_submission.rst:13]: Let's create an issue template directly on the github repo, so that we don't have to recreate it here :) This can be done here: https://github.com/PennyLaneAI/qml/issues/templates/edit

We can even have multiple issue templates, and it will look quite nice:

image

File: demos_submission.rst [hunk 2]
```diff
@@ -8,8 +8,101 @@ How to submit a demo
...
+
+Create a demo
+-------------
+Create a demonstration/tutorial that is using PennyLane and upload it to e.g.
+GitHub, Bitbucket or GitLab. Preferably, the demo is simply written as a Jupyter
+notebook or a Python document (using either Python comments or Restructured Text
+sections starting with 79 hashes ``#``), but you may write the demo in any way
+you like.
```
Comment (josh146) [inline demos_submission.rst:21]: Is this page directly related to community submissions? If so, with community submission, is the plan to be less strict with submissions?

E.g., submission can either be:

* A link to a GitHub/bitbucket/gitlab repository (c.f. https://github.com/PatrickHuembeli/vqc_loss_landscapes)

* A link to a rendered jupyter notebook (can be hosted anywhere, e.g., GitHub, colab, nbviewer)

Comment (josh146) [reply demos_submission.rst:21]: I suppose we need to delineate two types of submissions: ones that go into the basics/research tab, and ones that go into the community tab. The former requires a Python script (as well as instructions on how to build the site locally), the latter is less strict.

File: demos_submission.rst [hunk 3]
```diff
@@ -8,8 +8,101 @@ How to submit a demo
...
+* Code should be clearly commented and explained.
+
+* If your content contains random variables/outputs, a fixed seed should be set
+  for reproducibility.
+
+* All content should be original or free to reuse subject to license
+  compatibility. For example, if you are implementing someone else's research,
+  reach out first to recieve permission to reproduce exact figures. Otherwise,
+  avoid direct screenshots from papers, and instead refer to figures in the
+  paper within the text.
```
Comment (josh146) [inline demos_submission.rst:48]: Nice!

File: demos_submission.rst [hunk 4]
```diff
@@ -8,8 +8,101 @@ How to submit a demo
...
+
+Open a `new issue on the PennyLane QML GitHub repository
+<https://github.com/PennyLaneAI/qml/issues/new>`_ and paste the filled in
+template from above in the description box. Remove and replace all the text that
+was already there.
+
+Write "[DEMO] your-demo-title" (e.g. "[DEMO] Frugal shot optimization with
+Rosalin", without the quotes) as the title of the issue, and add the "demos"
+label to the issue by clicking on the cogwheel on the right-hand side of the
+**Labels** tag to the right of the text-box and then clicking on "demos".
```
Comment (josh146) [inline demos_submission.rst:102]: I think we can set the title as part of the github issue template?

File: demos_submission.rst [hunk 5]
```diff
@@ -8,8 +8,101 @@ How to submit a demo
...
+
+Don't forget to push the "Submit new issue" button!
```
Comment (josh146) [inline demos_submission.rst:104]: 😆

**Review by josh146 — COMMENTED** [review]
Thanks @thisac! Looks good, my main suggestions relate to delineating between the types of submissions.

File: demos_submission.rst (hunk 1 above)
Comment (thisac) [reply demos_submission.rst:13]: Thanks @josh146. This is a great idea! I've created new issue templates ([this PR](https://github.com/PennyLaneAI/qml/pull/138)).

File: demos_submission.rst (hunk 4 above)
Comment (thisac) [reply demos_submission.rst]: Yep, fixed! :tada:

File: demos_submission.rst (hunk 1 above)
Comment (josh146) [reply demos_submission.rst:13]: Amazing! Once it's merged, you can have a link here that takes you straight to the issue template to fill in:

https://github.com/PennyLaneAI/pennylane/issues/new?assignees=&labels=demo&template=community-demo.md

File: demos_submission.rst [hunk 6]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+
+    <blockquote class="blockquote border rounded ">
+        <p class="mb-0 text-muted text-center">
+            You can also sumbit a link to a rendered Jupyter notebook using e.g.
+            <a href="https://colab.research.google.com/notebooks/intro.ipynb">Colab</a>,
+            <a href="https://nbviewer.jupyter.org/">nbviewer</a>,
+            <a href="https://aws.amazon.com/sagemaker/">Sagemaker</a>
+            or <a href="https://mybinder.org/">Binder</a>.
+        </p>
+    </blockquote>
```
Comment (mariaschuld) [inline demos_submission.rst:31]: To be honest, I thought this formatting was a bug in sphinx. It wasn't so clear why this part of the text is really large...

Comment (thisac) [reply demos_submission.rst:31]: Oh, maybe it's a bit too much. Do you think it would be better just having it as text and not in a box? @mariaschuld

File: demos_community.rst [hunk 7]
```diff
@@ -8,15 +8,38 @@ Community
...
+
+        <div class="card-body text-center mb-3">
+            <p class="card-text py-2">
+                Have an existing GitHub repository or Jupyter notebook showing
+                off quantum machine learning with PennyLane? Read the guidelines
+                and submission instructions <a href="demos_submission.html">here</a>,
+                and have your demonstration and research featured on our community page.
+            </p>
+        </div>
+    </div>
```
Comment (josh146) [inline demos_community.rst:35]: I love this landing page vibe. I think it looks great!

A very minor suggestion: the text at the top of the page, as well as the text in the jumbotron, tend to repeat each other a bit. We could remove the top text (for now), and simply have the jumbotron:

image

In the screenshot, I've modified the CSS slightly to make the Jumbotron border less harsh, and add the class `"lead"` to the `card-text` paragraph:

```css
box-shadow: 0 2px 20px 0 rgba(0,0,0,.08),0 2px 50px 0 rgba(0,0,0,.08);
border-radius: 12px;
```

I suppose when we have our first community demos, we can remove the jumbotron, and go back to just having lead text at the top?

File: demos_submission.rst [hunk 8]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
    :property="og:description": Take a deeper dive into quantum machine learning by exploring cutting-edge algorithms using PennyLane and near-term quantum hardware.
    :property="og:image": https://pennylane.ai/qml/_static/demos_card.png
 
+To submit your own demo you can follow the steps below, beginning with creating
+a demo, followed by filling in a demo-submission template and posting it as an
+`issue to the PennyLane QML GitHub repository <https://github.com/PennyLaneAI/qml/issues/new?assignees=&labels=demos&template=community-demo.md&title=%5BDEMO%5D>`_
```
Comment (josh146) [inline demos_submission.rst:13]: 🖍️ ✒️

File: demos_submission.rst [hunk 9]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+
+Create a demo
+-------------
+Create a demonstration/tutorial that is using PennyLane and upload it to e.g.
+GitHub, Bitbucket or GitLab. Preferably, the demo is simply written as a Jupyter
+notebook or a Python document, but you may write the demo in any way you like.
```
Comment (josh146) [inline demos_submission.rst:19]: Can we make it clear as well that entire GitHub repos can count as a submission? I'm thinking in terms of something like https://github.com/PatrickHuembeli/vqc_loss_landscapes. GitHub repos with multiple files are a very common way of providing code for papers in ML!

File: demos_submission.rst (hunk 6 above)
Comment (josh146) [reply demos_submission.rst:31]: Yeah, I have to admit I found this stood out weirdly as well. Not so much the style, but the text it emphasizes.

"You can also..." sounds like an addendum rather than a main point, so I found myself asking why this particular paragraph was emphasized!

File: demos_submission.rst [hunk 10]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+        </p>
+    </blockquote>
+
+Guidelines
+^^^^^^^^^^
+
+While you are free to be as creative as you like with your demo, there are a
+couple of guidelines to keep in mind.
+
+* Submissions should include your name (and optionally email) at the top under
```
Comment (josh146) [inline demos_submission.rst:39]: Did this at one point have 'username' as well?

File: demos_submission.rst [hunk 11]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+            <a href="https://aws.amazon.com/sagemaker/">Sagemaker</a>
+            or <a href="https://mybinder.org/">Binder</a>.
+        </p>
+    </blockquote>
+
+Guidelines
+^^^^^^^^^^
+
+While you are free to be as creative as you like with your demo, there are a
+couple of guidelines to keep in mind.
```
Comment (josh146) [inline demos_submission.rst:37]: It might be good to clarify what these guidelines actually refer to. When I first read this, my reading was that they apply to the info that will be included on the card on the 'Community' page.

If they apply to the demo itself, it's not very clear where they apply if the demo is not in the form of a single Jupyter notebook. E.g., if linking to a GitHub repo of files, do the guidelines apply to the GitHub readme?

File: demos_submission.rst [hunk 12]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+  the title.
+
+* The title should be clear and concise, and if based on a paper it should be
+  similar to the paper that is being implemented.
+
+* All demos should include a summary below the title. The summary should be 1-3
+  sentences that makes clear the goal and outcome of the demo, and links to any
+  papers/resources used.
+
+* Code should be clearly commented and explained.
```
Comment (josh146) [inline demos_submission.rst:49]: Another guideline: it would be good to include the dependency versions require to run the demo successfully. E.g., it should say `pennylane==0.10.0` etc. This way, if the demo is broken on later versions, it's still easy to install an environment where everything works.

Comment (josh146) [reply demos_submission.rst:49]: And it will help us if users file bug reports!

Comment (josh146) [reply demos_submission.rst:49]: Perhaps, at minimum, the PL and plugin versions should be provided. Even better, a `requirements.txt` file with the output of `pip freeze` would be 👨‍🍳

File: demos_submission.rst [hunk 13]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+
+* All content should be original or free to reuse subject to license
+  compatibility. For example, if you are implementing someone else's research,
+  reach out first to receive permission to reproduce exact figures. Otherwise,
+  avoid direct screenshots from papers, and instead refer to figures in the
+  paper within the text.
+
+
+Open an issue
+-------------
```
Comment (josh146) [inline demos_submission.rst:62]: I wonder if this section is still necessary now that we have the github template? 🍕 4️⃣ 💭

File: xanadu_theme/layout.html [hunk 14]
```diff
@@ -168,7 +168,7 @@
           </div>
           {%- block comments -%}
 
-            {% if pagename in ["demos_basics", "demos_research", "demos_community"] %}
+            {% if pagename in ["demos_basics", "demos_research"] %}
```
Comment (josh146) [inline xanadu_theme/layout.html:171]: 😆 good catch

Comment (josh146) [reply xanadu_theme/layout.html:171]: Although, I suppose we could always add this back in at some point! Include a 'tags' version in the issue template.

**Review by josh146 — COMMENTED** [review]
Love the new community page @thisac! Once my comments are resolved, this will be good to be merged in on my side.

File: demos_submission.rst (hunk 6 above)
Comment (thisac) [reply demos_submission.rst:31]: Right, I'll de-emphasize it then. Not really sure why I boxed it in the first place; it just seemed like a good thing to do at the time. :laughing:

File: demos_community.rst (hunk 7 above)
Comment (thisac) [reply demos_community.rst:35]: This looks much better. Thanks @josh146!

File: demos_submission.rst (hunk 10 above)
Comment (thisac) [reply demos_submission.rst:39]: No, it only said name as far as I'm aware. I think this is fine here, or would you prefer writing `name or username` instead?

File: demos_submission.rst (hunk 12 above)
Comment (thisac) [reply demos_submission.rst:45]: Added a bit mentioning that dependencies could be added to the tutorial. Not sure if this would actually fit better in the issue rather than in the demo itself, although it's at least mentioned in the guidelines now.

File: demos_submission.rst (hunk 13 above)
Comment (thisac) [reply demos_submission.rst:62]: I don't mind this being here as well, as an example, although I agree that it might not be necessary. What do you think, should I just remove it or keep it as it is? :thinking: 

:pizza: :four: :thought_balloon: (that's a nice little rebus, by the way; didn't notice it at first)

File: _static/xanadu_gallery.css [hunk 15]
```diff
@@ -236,4 +236,9 @@ p.sphx-glr-signature a.reference.external {
   margin-top: 20px;
   text-align: center;
   font-size: small;
-}
\ No newline at end of file
+}
+
+.jumbotron {
+    box-shadow: 0 2px 20px 0 rgba(0,0,0,.08),0 2px 50px 0 rgba(0,0,0,.08) !important;
+    border-radius: 12px !important;
```
Comment (thisac) [inline _static/xanadu_gallery.css:243]: It wouldn't work without the `!important`.

File: demos_submission.rst (hunk 10 above)
Comment (josh146) [reply demos_submission.rst]: How it is is fine. I had a vague recollection of @co9olguy changing 'pseudonym' to 'username', but wasn't sure where that was

File: demos_submission.rst (hunk 13 above)
Comment (josh146) [reply demos_submission.rst:62]: >I agree that it might not be necessary

I don't really mind either way, my only thought was that it might be harder to maintain two separate copies in two separate files.

> 🍕 4️⃣ 💭

It did make me laugh when I included it lol. I might start using this instead of 'nitpick' to label suggested changes that aren't necessary

File: demos_submission.rst [hunk 16]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+To submit your own demo you can follow the steps below, beginning with creating
+a demo, followed by filling in a demo-submission template and posting it as an
+`issue to the PennyLane QML GitHub repository <https://github.com/PennyLaneAI/qml/issues/new?assignees=&labels=demos&template=community-demo.md&title=%5BDEMO%5D>`_
+
+Create a demo
+-------------
+Create a demonstration/tutorial that is using PennyLane and upload it to e.g.
+GitHub, Bitbucket or GitLab. Preferably, the demo is simply written as a Jupyter
+notebook a Python document or simply as a repository (e.g. `like this <https://github.com/PatrickHuembeli/vqc_loss_landscapes>`_), but you may
+write the demo in any way you like.
```
Comment (josh146) [inline demos_submission.rst:20]:
```suggestion
Create a demonstration/tutorial using PennyLane and upload it to a source code
hosting service such as GitHub, Bitbucket or GitLab. Good formats for a demonstration include:

* Jupyter notebooks [`example <https://github.com/lanabozanic/quantum_natural_gradient/blob/master/qng_tutorial.ipynb>__`],

* Python scripts with detailed comments and explanations
  [`example <https://github.com/PennyLaneAI/qml/blob/master/demonstrations/tutorial_vqe.py>`__],

* Repositories containing a collection of Python scripts, Jupyter notebooks, and
  documentation [`example1 <https://github.com/PatrickHuembeli/vqc_loss_landscapes>`__,
  `example2 <https://github.com/hsim13372/quantum_natural_gradient>`__].
```

Comment (josh146) [reply demos_submission.rst:20]: Adding examples is a great idea @thisac, I suddenly thought of a few more we could add for each style of demo. It might be good to get permission before going live though.

File: demos_submission.rst [hunk 17]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+
+Alternatively, you can also submit a link to a rendered Jupyter notebook using e.g. `Colab
+<https://colab.research.google.com/notebooks/intro.ipynb>`_, `nbviewer
+<https://nbviewer.jupyter.org/>`_, `Sagemaker <https://aws.amazon.com/sagemaker/>`_ or `Binder <https://mybinder.org/>`_.
+
+Guidelines
+^^^^^^^^^^
+
+While you are free to be as creative as you like with your demo, there are a
+couple of guidelines that might be good keep in mind when creating a demo.
```
Comment (josh146) [inline demos_submission.rst:30]: (to avoid repetition)
```suggestion
While you are free to be as creative as you like with your demo, there are a
couple of guidelines that might be good keep in mind.
```

File: demos_submission.rst [hunk 18]
```diff
@@ -8,8 +8,102 @@ How to submit a demo
...
+Specifically when it's a Jupyter notebook or a single Python document. The
+following concerns the content of the demo itself, and is in addition to the
+submission details explained in the next section.
```
Comment (josh146) [inline demos_submission.rst:33]: Maybe the following?

```suggestion
The following guidelines refer to the content of the demo itself, or, if submitting
a GitHub repository, the repository documentation/README file.
```

File: _static/xanadu_gallery.css (hunk 15 above)
Comment (josh146) [reply _static/xanadu_gallery.css:243]: @thisac, this should be in the `xanadu_theme/static/xanadu.css_t` file. This is the main css file (note the `_t`), and should avoid the need to use `!important` (I think?)

**Review by josh146 — APPROVED** [review]
Thanks @thisac! Left some final comments polishing the how-to section, once incorporated, I'm happy for this to be merged in.

File: demos_submission.rst (hunk 16 above)
Comment (thisac) [reply demos_submission.rst:20]: That's great. Thanks @josh146!

File: demos_submission.rst (hunk 10 above)
Comment (thisac) [reply demos_submission.rst]: Yes, that was in the issue template. :smile:

File: _static/xanadu_gallery.css (hunk 15 above)
Comment (thisac) [reply _static/xanadu_gallery.css]: @josh146 I moved it to the xanadu.css_t file (I think I missed it because my editor didn't highlight it as a css file :laughing: ), but it still needed `!important`. :thinking:

Comment (josh146) [reply _static/xanadu_gallery.css]: All good, we can keep the `!important`!

The reason for `.css_t` is that this is a _templated_ css file. Some of the values are simply template variables,

```css
.btn-xanadu {
  color: #ffffff;
  background-color: {{ theme_navigation_button }};
  border: none;
}
```

where the value of the template variable is specified in `conf.py`:
```python
# xanadu theme options (see theme.conf for more information)
html_theme_options = {
    "navigation_button": "#19b37b",
    "navigation_button_hover": "#0e714d",
    "toc_caption": "#19b37b",
    "toc_hover": "#19b37b",
    "table_header_bg": "#edf7f4",
    "table_header_border": "#19b37b",
    "download_button": "#19b37b",
```

This allows us to change specific elements of the theme globally (e.g., this is used to easily change the 'accent' colour from red/green/blue across SF/PennyLane/TheWalrus!)
